hw_07.09.23/task2/phonebook.py

```diff
@@ -0,0 +1,138 @@
+import os
+import json
+import argparse
+
+parser = argparse.ArgumentParser()
+parser.add_argument("data_path", help="Phonebook file path")
+args = parser.parse_args()
+
+current_dir = os.path.dirname(os.path.realpath(__file__))
+
+PHONEBOOK = {}
+
+datafile_path = os.path.join(current_dir, (args.data_path + ".json"))
+
+if os.path.exists(datafile_path):
+    with open(datafile_path, "r") as file:
+        PHONEBOOK = json.load(file)
+
+
+def add_contact():
+    first_name = input("Enter first name: ")
+    last_name = input("Enter last name: ")
+    full_name = first_name + " " + last_name
+    telephone_number = input("Enter telephone number: ")
+    city_or_state = input("Enter city or state: ")
+
+    if telephone_number and full_name:
+        PHONEBOOK[telephone_number] = {
+            "first_name": first_name,
+            "last_name": last_name,
+            "full_name": full_name,
+            "address": city_or_state,
+        }
+
+        with open(datafile_path, "w") as file_obj:
+            json.dump(PHONEBOOK, file_obj, indent=4)
+            print(PHONEBOOK)
+
+    else:
+        print("Invalid input. Both name and telephone number are required.")
+
+
+def search_by_fullname(type_: str) -> dict:
```

Comment:
A small nit: I'd recommend to rename `_type` -> `search_pattern` or `search_string` here and later. Not so important, of course